### PR TITLE
Add YAML configuration code samples

### PR DESCRIPTION
**Pull request by blazember**

### CODE REVIEW

File: learning-basics/configure-yaml/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- `ClasspathYamlConfig`: loads the configuration from the classpath based on the provided filename
+- `FileSystemYamlConfig`: loads the configuration from the file system based on the provided filename
+- `UrlYamlConfig`: loads the configuration from the provided URL
+- `InMemoryYamlConfig`: loads the configuration from the provided `String` containing YAML configuration 
+
+In the case of Java clients `ClientClasspathYamlConfig` can be used for the same purpose.
+  
+The sample `YamlConfigClasspath` loads the member configuration `hazelcast-sample.yaml` from the classpath and 
+demonstrates variable replacement in YAML configuration by defining the member's port in variable taken from 
+the system properties and then from a provided `Properties` instance.
```
Comment (leszko) [inline learning-basics/configure-yaml/README.md]: Would it make sense to add the information on how to run this sample and what the reader should observe?

File: learning-basics/configure-yaml/src/main/java/ConfigLookup.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.hazelcast.core.HazelcastInstance;
+
+import static com.hazelcast.examples.helper.CommonUtils.sleepMillis;
+
+public class ConfigLookup {
+    public static void main(String[] args) {
+        // XML takes precedence: loading member configuration hazelcast.xml from the classpath
+        // despite the presence of hazelcast.yaml
+        HazelcastInstance member = Hazelcast.newHazelcastInstance();
+        sleepMillis(1000);
```
Comment (leszko) [inline learning-basics/configure-yaml/src/main/java/ConfigLookup.java]: Why do you need this sleeps? For the user to observe the results? Or this is necessary for some reason? Could you maybe comment it in the code?

File: learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+hazelcast-client:
+  import:
```
Comment (leszko) [inline learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml]: I guess this import is for the reader to observe something, would it make sense to describe it in README?

File: learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    - hazelcast-client-network-config.yaml
+  group:
+    name: dev-configured-from-yaml
```
Comment (leszko) [inline learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml]: Do you use it somehow in the code sample?

File: learning-basics/configure-yaml/README.md (hunk 1 above)
Comment (blazember) [reply learning-basics/configure-yaml/README.md]: The samples are implemented in main methods without using shell scripts. I can add that, but I think it's not a question for a java engineer.

What the user should observe when running the samples is added as inline comments.

File: learning-basics/configure-yaml/src/main/java/ConfigLookup.java (hunk 2 above)
Comment (blazember) [reply learning-basics/configure-yaml/src/main/java/ConfigLookup.java]: I used the `configuration-xml` sample as a basis, the sleeps were there. Removed, it wasn't necessary.

File: learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml (hunk 3 above)
Comment (blazember) [reply learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml]: Added to the README that the first test demonstrates composing YAML configuration files.

File: learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml (hunk 4 above)
Comment (blazember) [reply learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml]: It is needed to be the same in member and client config. It's better to be unique in the samples so no members and clients join from or to instances started by other samples by mistake.

**Comment by blazember** [comment]
Thanks for the comments @leszko, addressed them. Could you please take a second look?

**Comment by blazember** [comment]
Thanks for the review @lazerion @leszko :+1: 

Squashed and added a new section referencing the `hazelcast-full-example.yaml` configuration to the README.

**Pull request merged**